### PR TITLE
APIMF-3048: add javadoc and remove private[amf]

### PR DESCRIPTION
**Pull request by arielmirra**
- Add javadoc to most remodularization classes, methods and objects
- Remove the private[amf] modifier on those classes/methods/objects

### CODE REVIEW

File: shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala [hunk 1]
```diff
@@ -28,10 +28,14 @@ private[amf] object AMFGraphConfiguration {
 
   /**
     * Predefined AMF core environment with
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala]: `with:` add colons

Comment (tomsfernandez) [reply shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala]: Maybe add bullets also if possible

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFOptions.scala [hunk 2]
```diff
@@ -1,5 +1,10 @@
 package amf.client.remod.amfcore.config
 
+/**
+  * A set of options to customize parsing and rendering
+  * @param parsingOptions {@link amf.client.remod.amfcore.config.ParsingOptions}
+  * @param renderOptions {@link amf.client.remod.amfcore.config.RenderOptions}
+  */
 private[amf] case class AMFOptions(parsingOptions: ParsingOptions,
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/config/AMFOptions.scala:8]: Should this be private ?

File: shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala [hunk 3]
```diff
@@ -121,11 +133,11 @@ private[amf] class AMFGraphConfiguration(override private[amf] val resolvers: AM
 
   }
 
-  private[amf] def getParsingOptions: ParsingOptions        = options.parsingOptions
-  private[amf] def getRegistry: AMFRegistry                 = registry
-  private[amf] def getResourceLoaders: List[ResourceLoader] = resolvers.resourceLoaders
-  private[amf] def getUnitsCache: Option[UnitCache]         = resolvers.unitCache
-  private[amf] def getExecutionContext: ExecutionContext    = resolvers.executionContext.executionContext
+  def getParsingOptions: ParsingOptions        = options.parsingOptions
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/AMFGraphConfiguration.scala]: Why isn't this private?

File: shared/src/main/scala/amf/client/remod/amfcore/config/ParsingOptions.scala [hunk 4]
```diff
@@ -4,38 +4,32 @@ import amf.core.client.{ParsingOptions => LegacyParsingOptions}
...
 
-  def isAmfJsonLdSerilization: Boolean   = amfJsonLdSerialization
+  def isAmfJsonLdSerialization: Boolean  = amfJsonLdSerialization
   def definedBaseUrl: Option[String]     = baseUnitUrl
   def getMaxYamlReferences: Option[Long] = maxYamlReferences
 
 }
 
-private[amf] object ParsingOptionsConverter {
+object ParsingOptionsConverter {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/config/ParsingOptions.scala]: This should be private

File: shared/src/main/scala/amf/client/remod/amfcore/plugins/render/AMFRenderPlugin.scala [hunk 5]
```diff
@@ -7,17 +7,20 @@ import amf.core.errorhandling.ErrorHandler
...
-  def emit[T](unit: BaseUnit, builder: DocBuilder[T], renderOptions: RenderOptions, errorHandler: ErrorHandler): Boolean
+trait AMFRenderPlugin extends AMFPlugin[RenderInfo] {
+  def emit[T](unit: BaseUnit,
+              builder: DocBuilder[T],
+              renderOptions: RenderOptions,
+              errorHandler: ErrorHandler): Boolean
 }
 
-private[amf] case class AMFRenderPluginAdapter(plugin: AMFDocumentPlugin) extends AMFRenderPlugin {
+case class AMFRenderPluginAdapter(plugin: AMFDocumentPlugin) extends AMFRenderPlugin {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/plugins/render/AMFRenderPlugin.scala]: This should be private

File: shared/src/main/scala/amf/client/remod/amfcore/registry/AMFRegistry.scala [hunk 6]
```diff
@@ -6,10 +6,17 @@ import amf.client.remod.amfcore.plugins.parse.{AMFParsePlugin, DomainParsingFall
...
-                                    transformationPipelines: Map[String, ResolutionPipeline],
-                                    constraintsRules: Map[ProfileName, ValidationProfile]) {
+/**
+  * Registry to store plugins, entities, transformation pipelines and constraint rules
+  * @param plugins {@link amf.client.remod.amfcore.registry.PluginsRegistry}
+  * @param entitiesRegistry {@link amf.client.remod.amfcore.registry.EntitiesRegistry}
+  * @param transformationPipelines a map of {@link amf.core.resolution.pipelines.ResolutionPipeline}s
+  * @param constraintsRules a map of {@link amf.ProfileName} -> {@link amf.core.validation.core.ValidationProfile}
+  */
+case class AMFRegistry private[amf] (plugins: PluginsRegistry,
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/amfcore/registry/AMFRegistry.scala]: This should be private

**Review by tomsfernandez — CHANGES_REQUESTED** [review]

**Comment by tomsfernandez** [comment]
Added @pope1838 as reviewer

File: shared/src/main/scala/amf/client/remod/amfcore/config/ParsingOptions.scala (hunk 4 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/ParsingOptions.scala]: Yeap. In fact,  conversion should be implicit

File: shared/src/main/scala/amf/client/remod/amfcore/config/AMFOptions.scala (hunk 2 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFOptions.scala:8]: The constructor should be private for sure. The class could be public I think. 
(If makes sense to have a withParsingOptions(options) at the configuration)

Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/amfcore/config/AMFOptions.scala:8]: I I think it should be private. Parsing and Emitting options could be public, but not this one as is just a container of the other two, and overriding the value could end overriding options previously set by default ones, without notice.

File: shared/src/main/scala/amf/client/remod/ErrorHandlerProvider.scala [hunk 7]
```diff
@@ -3,7 +3,7 @@ package amf.client.remod
 import amf.client.parse.DefaultParserErrorHandler
 import amf.core.errorhandling.ErrorHandler
 
-trait ErrorHandlerProvider {
+private[amf] trait ErrorHandlerProvider {
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/client/remod/ErrorHandlerProvider.scala]: This shouldn't be private. @pope1838

**Review by tomsfernandez — CHANGES_REQUESTED** [review]

File: shared/src/main/scala/amf/client/remod/ErrorHandlerProvider.scala (hunk 7 above)
Comment (pope1838) [reply shared/src/main/scala/amf/client/remod/ErrorHandlerProvider.scala]: You are right. The adopter could register an ErrorHandlerProvider. Is a client interface

**Pull request merged**